src/write_influxdb_udp.c

```diff
@@ -0,0 +1,701 @@
+  cdtime_t next_resolve_reconnect;
+  cdtime_t resolve_interval;
+  struct sockaddr_storage *bind_addr;
+};
+
+typedef struct sockent {
+  char *node;
+  char *service;
+  int interface;
+  struct sockent_client client;
+} sockent_t;
+
+#define NET_DEFAULT_PACKET_SIZE 1452
+#define NET_DEFAULT_PORT "8089"
+
+/*
+ * Private variables
+ */
+
+static int wifxudp_config_ttl;
+static size_t wifxudp_config_packet_size = NET_DEFAULT_PACKET_SIZE;
+static bool wifxudp_config_store_rates;
+
+static sockent_t *sending_socket;
+
+/* Buffer in which to-be-sent network packets are constructed. */
+static char *send_buffer;
+static char *send_buffer_ptr;
+static int send_buffer_fill;
+static cdtime_t send_buffer_last_update;
+static value_list_t send_buffer_vl = VALUE_LIST_INIT;
+static pthread_mutex_t send_buffer_lock = PTHREAD_MUTEX_INITIALIZER;
+
+static int listen_loop;
+
+static int set_ttl(const sockent_t *se, const struct addrinfo *ai) {
+
+  if ((wifxudp_config_ttl < 1) || (wifxudp_config_ttl > 255))
+    return -1;
+
+  if (ai->ai_family == AF_INET) {
+    struct sockaddr_in *addr = (struct sockaddr_in *)ai->ai_addr;
+    int optname;
+
+    if (IN_MULTICAST(ntohl(addr->sin_addr.s_addr)))
+      optname = IP_MULTICAST_TTL;
+    else
+      optname = IP_TTL;
+
+    if (setsockopt(se->client.fd, IPPROTO_IP, optname, &wifxudp_config_ttl,
+                   sizeof(wifxudp_config_ttl)) != 0) {
+      ERROR("network plugin: setsockopt (ipv4-ttl): %s", STRERRNO);
+      return -1;
+    }
+  } else if (ai->ai_family == AF_INET6) {
+    /* Useful example:
+     * http://gsyc.escet.urjc.es/~eva/IPv6-web/examples/mcast.html */
+    struct sockaddr_in6 *addr = (struct sockaddr_in6 *)ai->ai_addr;
+    int optname;
+
+    if (IN6_IS_ADDR_MULTICAST(&addr->sin6_addr))
+      optname = IPV6_MULTICAST_HOPS;
+    else
+      optname = IPV6_UNICAST_HOPS;
+
+    if (setsockopt(se->client.fd, IPPROTO_IPV6, optname, &wifxudp_config_ttl,
+                   sizeof(wifxudp_config_ttl)) != 0) {
+      ERROR("network plugin: setsockopt(ipv6-ttl): %s", STRERRNO);
+      return -1;
+    }
+  }
+
+  return 0;
+} /* int set_ttl */
+
+static int bind_socket_to_addr(sockent_t *se, const struct addrinfo *ai) {
+
+  if (se->client.bind_addr == NULL)
+    return 0;
+
+  char pbuffer[64];
+
+  if (ai->ai_family == AF_INET) {
+    struct sockaddr_in *addr = (struct sockaddr_in *)(se->client.bind_addr);
+    inet_ntop(AF_INET, &(addr->sin_addr), pbuffer, 64);
+
+    if (bind(se->client.fd, (struct sockaddr *)addr, sizeof(*addr)) == -1)
+      return -1;
+  } else if (ai->ai_family == AF_INET6) {
+    struct sockaddr_in6 *addr = (struct sockaddr_in6 *)(se->client.bind_addr);
+    inet_ntop(AF_INET6, &(addr->sin6_addr), pbuffer, 64);
+
+    if (bind(se->client.fd, (struct sockaddr *)addr, sizeof(*addr)) == -1)
+      return -1;
+  }
+
+  return 0;
+} /* int bind_socket_to_addr */
+
+static sockent_t *sockent_create() {
+  sockent_t *se;
+
+  se = calloc(1, sizeof(*se));
```

Comment:
New codestyle expects variables declared on first use, if possible.

Reply:
Thank you. It should be done.